src/controller/python/chip/setup_payload/Parser.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <setup_payload/ManualSetupPayloadParser.h>
+#include <setup_payload/QRCodeSetupPayloadParser.h>
+#include <support/CodeUtils.h>
+
+#include <string>
+
+using namespace chip;
+
+namespace {
+
+using AttributeVisitor       = void (*)(const char * attrName, const char * attrValue);
+using VendorAttributeVisitor = void (*)(uint8_t tag, const char * attrValue);
+
+void yieldSetupPayloadAttributes(const SetupPayload & payload, AttributeVisitor attrVisitor,
+                                 VendorAttributeVisitor vendorAttrVisitor)
+{
+    attrVisitor("Version", std::to_string(payload.version).c_str());
+    attrVisitor("VendorID", std::to_string(payload.vendorID).c_str());
+    attrVisitor("ProductID", std::to_string(payload.productID).c_str());
+    attrVisitor("RequiresCustomFlow", std::to_string(payload.requiresCustomFlow).c_str());
+    attrVisitor("RendezvousInformation", std::to_string(payload.rendezvousInformation.Raw()).c_str());
```

Comment:
Would be good to address #4936 right away.

Reply:
Right, done